server/src/operators/model_operator.rs

```diff
@@ -743,6 +741,77 @@ pub async fn cross_encoder(
         resp.into_iter().for_each(|pair| {
             results.index_mut(pair.index).score = pair.score as f64;
         });
+    } else {
+        let vec_futures: Vec<_> = request_docs
+            .chunks(20)
+            .enumerate()
+            .map(|(i, docs_chunk)| {
+                let clipped_docs = docs_chunk
+                    .iter()
+                    .map(|doc| {
+                        if doc.len() > 7000 {
+                            doc.chars().take(20000).collect()
+                        } else {
+                            doc.clone()
+                        }
+                    })
+                    .collect::<Vec<String>>();
+
+                let parameters = CrossEncoderData {
+                    query: query.clone(),
+                    texts: clipped_docs,
+                    truncate: true,
+                };
+
+                let cur_client = reqwest_client.clone();
+                let url = embedding_server_call.clone();
+                let embedding_api_key = get_env!("OPENAI_API_KEY", "OPENAI_API should be set");
+
+                async move {
+                    let embeddings_resp = cur_client
+                        .post(&url)
+                        .header("Authorization", &format!("Bearer {}", &embedding_api_key))
+                        .header("api-key", &embedding_api_key.to_string())
+                        .header("Content-Type", "application/json")
+                        .json(&parameters)
+                        .send()
+                        .await
+                        .map_err(|_| {
+                            ServiceError::BadRequest(
+                                "Failed to send message to embedding server".to_string(),
+                            )
+                        })?
+                        .text()
+                        .await
+                        .map_err(|_| {
+                            ServiceError::BadRequest(
+                                "Failed to get text from embeddings".to_string(),
+                            )
+                        })?;
+
+                    let embeddings: Vec<ScorePair> = serde_json::from_str(&embeddings_resp)
+                        .map_err(|e| {
+                            log::error!("Failed to format response from embeddings server {:?}", e);
+                            ServiceError::InternalServerError(
+                                "Failed to format response from embeddings server".to_owned(),
+                            )
+                        })?;
```

Comment:
This request does not match the ureq request in terms of format. One specific example would be that it should be `.into_json::<Vec<ScorePair>>()` instead of `text()`.